code/modules/mob/living/carbon/xenomorph/death.dm

```diff
@@ -54,8 +54,17 @@
 						break
 				for(var/mob/living/carbon/Xenomorph/L in hive.xeno_leader_list)
 					L.handle_xeno_leader_pheromones(XQ)
-				if(ticker && ticker.mode)
-					ticker.mode.check_queen_status(hive.queen_time)
+				if(ticker?.mode)
+					var/i = 0
+					for(var/mob/living/carbon/Xenomorph/X in GLOB.alive_xeno_list)
```

Comment:
```suggestion
					for(var/X in GLOB.alive_xeno_list)
```
Guess it's more efficient this way, else it will do an `istype` check every time, when we know the list only has xenos.